### PR TITLE
Add feature set add / delete to repo manager

### PR DESCRIPTION
**Pull request by dcolligan**
Issue #1109

### CODE REVIEW

**Comment by jeromekelleher** [comment]
Looks good @dcolligan, thanks. I can't figure out how this relates to PRs #1104 and #1107 though... what have you done here?

Verify functionality would be good here. Just iterate over each FeatureSet and pull a few features out of it to confirm it's all good.

**Comment by dcolligan** [comment]
Doesn't relate at all to #1107

Pulls a few functions out of #1104

**Comment by jeromekelleher** [comment]
We really need to merge #1104 ASAP...

**Comment by dcolligan** [comment]
I agree, but it's not my +1 that you're waiting for!

**Comment by dcolligan** [comment]
@jeromekelleher updated

File: ga4gh/cli.py [hunk 1]
```diff
@@ -1531,6 +1532,15 @@ def __init__(self, args):
         self._repoPath = args.repoPath
         self._repo = datarepo.SqlDataRepository(self._repoPath)
 
+    def _getDatasetFromArgs(self):
+        try:
```
Comment (jeromekelleher) [inline ga4gh/cli.py]: This isn't necessary --- the exception should be propagated up automatically and caught at the CLI. This should be working nicely for (say) ReadGroupSets in the current sql repo code.

File: ga4gh/cli.py [hunk 2]
```diff
@@ -1682,6 +1692,33 @@ def func():
...
+        """
+        Adds a new feature set into this repo
+        """
+        self._openRepo()
+        dataset = self._getDatasetFromArgs()
+        name = getNameFromPath(self._args.filePath)
+        featureSet = sequenceAnnotations.Gff3DbFeatureSet(
+            dataset, name)
+        referenceSet = self._repo.getReferenceSetByName(
+            self._args.referenceSetName)
```
Comment (jeromekelleher) [inline ga4gh/cli.py]: What happens when ``referenceSetName`` is None? Have we any hope of guessing the correct reference set from a given FeatureSet? We should output an explicit error message here anyway.

**Comment by jeromekelleher** [comment]
Looks good @dcolligan, some minor comments above. We should make sure we error out nicely in all possible situations now; this is really important for a good user experience when dealing with confusing and difficult data.

**Comment by dcolligan** [comment]
@jeromekelleher updated

File: ga4gh/datarepo.py [hunk 3]
```diff
@@ -345,6 +345,14 @@ def verify(self):
                     reference.getLocalId())
         for dataset in self.getDatasets():
             print("Verifying Dataset", dataset.getLocalId())
+            reference = self.getReferenceSets()[0].getReferences()[0]
```
Comment (jeromekelleher) [inline ga4gh/datarepo.py]: This is broken --- you need to get the referenceSet for each of the FeatureSets. No reason to think that they'd all be the same.

**Comment by jeromekelleher** [comment]
Looks good, +1 after the reference issue in ``verify`` is fixed.

**Comment by dcolligan** [comment]
@jeromekelleher updated

**Comment by jeromekelleher** [comment]
Thanks, merging.

**Pull request merged**